### ISSUE TITLE
[Inductor] Illegal TMA store due to missing `tma_min_block_sizes`

### ISSUE THREAD

**Issue by liqiangxl** [pytorch/pytorch#190363]
## Issue

The following split reduction fails:

```python
import torch
from torch._inductor import config

n = 1 << 20
with config.patch({"triton.use_tensor_descriptor": True, "assume_aligned_inputs": True}):
    base = torch.randn((1, 2 * n), device="cuda")
    a = torch.as_strided(base, (1, n), (2 * n, 2))  # stride-2 reduction dim
    torch.compile(torch.sum, fullgraph=True)(a, dim=-1)  # crashes
```

The error:

```
Descriptor block shape must have at least 16 bytes in the last dimension,
but got 1 * 4 = 4 bytes
```

The reduction is split and its first kernel computes partial sums over chunks of the reduction and stores them to an intermediate
buffer via a TMA descriptor:

```python
def triton_per_fused_sum_0(in_ptr0, out_ptr0, xnumel, r0_numel, XBLOCK : tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:, None]
    ...
    tl.make_tensor_descriptor(out_ptr0, shape=[8192], strides=[1], block_shape=[XBLOCK]).store(
        [xoffset], tl.reshape(tl.broadcast_to(tmp3, [XBLOCK, 1]), [XBLOCK]).to(tl.float32)
    )
```

The descriptor requires `XBLOCK * 4 >= 16` (i.e. `XBLOCK >= 4` for float32),
but the runtime admits `XBLOCK=1` because `tma_min_block_sizes` was stripped
from the kernel metadata.

The bug also fires with cooperative (grid) reduction when `xnumel > 1`:

```python
with config.patch({
    "triton.use_tensor_descriptor": True,
    "assume_aligned_inputs": True,
    "split_reductions": False,
    "triton.cooperative_reductions": True,
    "triton.force_cooperative_reductions": True,
}):
    base = torch.randn((4, 2 * n), device="cuda")
    a = torch.as_strided(base, (4, n), (2 * n, 2))
    torch.compile(torch.sum, fullgraph=True)(a, dim=-1)  # same crash
```

## Cause

Two store paths in `torch/_inductor/codegen/triton.py`:

| Path | Sets `_emitted_device_tma`? |
|------|-----------------------------|
| `TritonKernel.store` (line ~4838) | ✅ via `codegen_block_ptr()` → line 4267 |
| `TritonKernel.store_reduction` (line ~5897) | ❌ calls `codegen_block_ptr_store_line()` directly |

`store_reduction` builds the descriptor with `indexing.format(var)` (returns
`tl.make_tensor_descriptor(...)`) and passes it straight to
`codegen_block_ptr_store_line`, bypassing `codegen_block_ptr` — the only place
that sets `self._emitted_device_tma = True`.

A cleanup guard in `codegen_kernel()` then strips the constraint because the
flag is still `False`:

```python
# triton.py ~7057-7063
if (
    not self.inductor_meta.get("host_tma_descriptor_args")
    and not self._emitted_device_tma
):
    self.inductor_meta.pop("tma_min_block_sizes", None)
```

This removes `{"XBLOCK": 4}` that `TMACompatibilityChecker` had correctly
recorded, so `_maybe_filter_configs_for_tma_restrictions()` never filters out
illegal configs.

The bug only surfaces when `store_reduction` is the *sole* TMA path in the
kernel — i.e. the input load is not TMA-compatible (stride ≠ 1 on the reduction
dim).  If the same kernel also has a TMA load through `codegen_block_ptr`, the
flag gets set there and masks the problem.

## Fix

Set `_emitted_device_tma` in `store_reduction` when emitting a TMA descriptor:

```python
if isinstance(indexing, (BlockPtrOptions, TensorDescriptorOptions)):
    if isinstance(indexing, TensorDescriptorOptions):
        self._emitted_device_tma = True
    self.post_loop_store.writeline(
        DeferredLine(
            name,
            self.codegen_block_ptr_store_line(
                name, indexing, indexing.format(var), value,
                f", boundary_check={indexing.boundary_check()!r}",
            ),
        )
    )
```

### Versions

main branch commit 4de523d0df93253de07539557a5006ae88d4ec8f (origin/main, main)
[pip3] torch==2.14.0a0+git6ec609c
[pip3] torch_c_dlpack_ext==0.1.5
[pip3] triton==3.7.1+git5d6048aa

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by liqiangxl** [comment]
These 2 tests failed on GB200 due to this issue and should be fixed after https://github.com/pytorch/pytorch/pull/190365
```
test/inductor/test_torchinductor_strided_blocks.py::TritonTensorDescriptorTestCUDA::test_welford_non_block_pointer_cuda
test/inductor/test_torchinductor_strided_blocks.py::TritonHostSideTMATestCUDA::test_welford_non_block_pointer_cuda```

**Comment by speedyshreya** [comment]
I'd like to work on this, could you please assign it to me?
